pommerman/envs/graphics.py

```diff
@@ -0,0 +1,175 @@
+
+# A map between the np value (Item(Enum) class)
+# to the position in the sprite array (ImageGrid).
+#
+# prob a better way to do this...
+IMAGE_FILE_NAMES = [
+    'Sprites-01.png', # Passage
+    'Sprites-02.png', # Rigid
+    'Sprites-03.png', # Wood
+    'Sprites-04.png', # Bomb
+    'Sprites-09.png', # Flames
+    'Sprites-10.png', # Fog
+    'Sprites-05.png', # ExtraBomb
+    'Sprites-06.png', # IncrRange
+    'Sprites-07.png', # Kick
+    'Sprites-08.png', # Skull
+    'Sprites-11.png', # AgentDummy -- Needs sprite
+    'Sprites-11.png', # Agent0
+    'Sprites-12.png', # Agent1
+    'Sprites-13.png', # Agent3
+    'Sprites-14.png', # Agent3
+]
+
+TILE_SIZE = 50
+IMAGE_DIR = 'images/'
+
+def empty_frame(board_size):
+    return [[None]*board_size for _ in range(board_size)]
+
+def index_images():
+    # Tell pyglet where to find the resources
+    __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+    _imagePath = os.path.join(__location__,  IMAGE_DIR)
+    pyglet.resource.path = [_imagePath]
+    pyglet.resource.reindex()
+
+
+class PommeViewer(object):
+    def __init__(self, display=None, board_size=13, agents=[], partially_observable=False):
+        display = rendering.get_display(display)
+
+        height = TILE_SIZE * board_size
+        width = height + height / 4
+
+        self._height = height
+        self._width = width
+        self._frame_section = []
+        self.window = pyglet.window.Window(width=width, height=height, display=display)
+        self.isopen = True
+        self._board_size = board_size
+        self._sprite_manager = SpriteManager()
+        self._tile_size = TILE_SIZE
+        self._agent_tile_size =  (height / 4) / board_size
+        self._agent_count = len(agents)
+        self._agents = agents
+        self._is_partially_observable = partially_observable == True
+        self._agent_view_size = 4
+
+        glEnable(GL_BLEND)
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA)
+
+        @self.window.event
+        def close(self):
+            self.window.close()
+            self.isopen = False
+
+    def set_agent_view_size(self, view_size):
+        self._agent_view_size =view_size
+
+    def set_frame_section(self, frames):
+        self._frame_section = frames
+    
+    def clear_frame_section(self):
+        self._frame_section = []
+
+    def set_agents_state(self, agents):
+        self._agents = agents
+
+    def set_frames_from_game_state(self, state):
+        images = empty_frame(self._board_size)
+
+        for r in range(self._board_size):
+            for c in range(self._board_size):
+                item_value = state[r][c]
+                image = self._sprite_manager.get_image_from_state_value(item_value)
+                images[r][c] = image
+        
+        self.set_frame_section(images)
+
+    def render(self):
+        self.window.clear()
+        self.window.switch_to()
```

Comment:
When there's just one window, `switch_to` is not needed. It doesn't harm to have it, but even then, it should come before `clear`, so that `clear` knows which window it should clear.